simpledb/compaction_test.go

```diff
@@ -142,3 +143,81 @@ func TestExecCompactionWithTombstoneRewriten(t *testing.T) {
 	// check size of compacted sstable
 	assert.Equal(t, 1000, int(db.sstableManager.currentSSTable().MetaData().NumRecords))
 }
+
+// regression reported in https://github.com/thomasjungblut/go-sstables/issues/36
+// more specifically this relates to a case where the first table is above the compaction threshold and never compacted.
+// The keyspace might already be partially rewritten, thus wasting unnecessary space.
+// This test is supposed to fail for later changes, thus the assertion is inverted
+func TestCompactionWithTombstonesBeyondMaxSize(t *testing.T) {
```

Comment:
@sebheitzmann I've added a regression test for this now - I hope this is the actual case you're referring to